### PR TITLE
Use GCM sink if SINK=gcm is specified.

### LINKED ISSUE #117: Add deployment support for the new GCM driver.
`deploy/run.sh` needs to be updated to support the new gcm backend driver.

### PR DESCRIPTION
**Pull request by vmarmol**
Default is influxdb which mirrors today's behavior.

Fixes #117.

### CODE REVIEW

**Comment by vmarmol** [comment]
Tested on a new cluster and it comes up as expected.

File: deploy/run.sh [hunk 1]
```diff
@@ -12,22 +12,37 @@ if [ ! -x $CADVISOR_PORT ]; then
...
 HEAPSTER="/usr/bin/heapster $EXTRA_ARGS "
-# Check if InfluxDB service is running
-if [ ! -z $KUBERNETES_RO_SERVICE_HOST ]; then
-  # TODO(vishh): add support for passing in user name and password.    
-  INFLUXDB_ADDRESS=""
-  if [ ! -z $MONITORING_INFLUXDB_SERVICE_HOST ]; then
-    INFLUXDB_ADDRESS="${MONITORING_INFLUXDB_SERVICE_HOST}:${MONITORING_INFLUXDB_SERVICE_PORT}"
+
+if [ "$SINK" == "influxdb" ]; then
+  # Check if InfluxDB service is running
```
Comment (vishh) [inline deploy/run.sh]: nit: The comment here should be "check if we are running in kubernetes". i see that it was incorrect to begin with.

File: deploy/run.sh [hunk 2]
```diff
@@ -12,22 +12,37 @@ if [ ! -x $CADVISOR_PORT ]; then
...
-  else 
-    echo "InfluxDB service address not found. Exiting."
-    exit 1
+    $HEAPSTER --sink influxdb --sink_influxdb_host ${INFLUXDB_HOST}
+  else
+    $HEAPSTER
+  fi
+elif [ "$SINK" == "gcm" ]; then
+  if [ ! -z $KUBERNETES_RO_SERVICE_HOST ]; then
+    $HEAPSTER --kubernetes_master "${KUBERNETES_RO_SERVICE_HOST}:${KUBERNETES_RO_SERVICE_PORT}" --sink gcm
```
Comment (vishh) [inline deploy/run.sh]: Should we do the master service discovery ahead of time and prepare heapster args before checking for storage drivers?

**Comment by vishh** [comment]
I think its time to rewrite `run.sh` in golang. May be after this PR.

File: deploy/run.sh (hunk 2 above)
Comment (vmarmol) [inline deploy/run.sh]: Done, looks cleaner.

File: deploy/run.sh (hunk 1 above)
Comment (vmarmol) [inline deploy/run.sh]: Done.

**Comment by vmarmol** [comment]
+1, this logic can probably live in the Heapster binary itself. We don't *need* a wrapper per say.

File: deploy/run.sh [hunk 3]
```diff
@@ -12,22 +12,42 @@ if [ ! -x $CADVISOR_PORT ]; then
...
+  SINK="influxdb"
+fi
+
+HEAPSTER="/usr/bin/heapster $EXTRA_ARGS "
+
+if [ "$SINK" == "influxdb" ]; then
+  HEAPSTER="$HEAPSTER --sink influxdb"
+
+  # Check if in Kubernetes.
+  if [ ! -z $KUBERNETES_RO_SERVICE_HOST ]; then
```
Comment (vishh) [inline deploy/run.sh]: I think we can get rid of this check.

File: deploy/run.sh [hunk 4]
```diff
@@ -12,22 +12,42 @@ if [ ! -x $CADVISOR_PORT ]; then
...
+    # TODO(vishh): add support for passing in user name and password.
+    INFLUXDB_ADDRESS=""
+    if [ ! -z $MONITORING_INFLUXDB_SERVICE_HOST ]; then
+      INFLUXDB_ADDRESS="${MONITORING_INFLUXDB_SERVICE_HOST}:${MONITORING_INFLUXDB_SERVICE_PORT}"
+    elif [ ! -z $INFLUXDB_HOST ]; then
+      INFLUXDB_ADDRESS=${INFLUXDB_HOST}
+    else
+      echo "InfluxDB service address not found. Exiting."
```
Comment (vishh) [inline deploy/run.sh]: nit: InfluxDB service address not specified.

File: deploy/run.sh (hunk 3 above)
Comment (vishh) [inline deploy/run.sh]: Ahh ignore my comment. I see below why its necessary.

**Comment by vishh** [comment]
LGTM except for the one nit. If we are moving to a dynamic configuration model, we can get rid of this logic completely.

**Comment by vishh** [comment]
@vmarmol: Feel free to self merge after fixing the nit.

File: deploy/run.sh (hunk 4 above)
Comment (vmarmol) [inline deploy/run.sh]: Done

**Comment by vmarmol** [comment]
Thanks @vishh! Will merge on green.

**Pull request merged**